### ISSUE TITLE
Geohash Integer methods

### ISSUE THREAD

**Issue by mablae** [thephpleague/geotools#56]
It would be quite nice to allow encoding into geohash integers. 

I need this functionalitly, would you accept PR on this featue? 

cheers mablae

**Comment by mablae** [comment]
@toin0u Any opinion on this?

**Comment by toin0u** [comment]
Hi @mablae! Sorry to reply so late. I think I'll go to refactor the current Geohash class :) Are you willing to help with this?

**Comment by mablae** [comment]
Sure, I am. I already looked at the current implementation and noticed that Geohash maintains its state internally as geohash string. How would you refactor the class in a sane way?

**Comment by mablae** [comment]
For a personal project I converted the algorithm used by https://github.com/sunng87/node-geohash
to PHP. 

Since there is also #3 (Geohash36) maybe it would be the better way to have a GeoHashInterface and then multiple self containing implementations on that? 

This would be easier to extend and no BC breaks at all.. 
Let's not couple all different implementations into the GeoHash class.

Converting between different geohashes would also be easily possible then... 

What do you think?

**Comment by toin0u** [comment]
@mablae I agree with you :) The interface is a good idea. I'm :+1:  for that.

Thanks for pointing out the issue with the long integer.. I think you're right to return the string make sense in this case.

Thank you very much for commenting and come with inputs, I appreciate it :)

**Comment by mablae** [comment]
Okay, fine. :+1: 

I will prepare a PR on the weekend to be discussed further then.

**Comment by toin0u** [comment]
That's sound nice! I'm looking forward to it :)

**Comment by toin0u** [comment]
@mablae any news about this? :)

**Comment by TeaDove** [comment]
Any news about integer version of Geohash?
